frontend/src/app/App.test.tsx

```diff
@@ -230,311 +239,382 @@ describe("App", () => {
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "SET_MENU_ITEMS",
+            items: [{ type: "separator" }],
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
 
-  it("updates state.appPages when it receives a PagesChanged msg", () => {
-    const appPages = [
-      { icon: "", pageName: "bob", scriptPath: "bob.py" },
-      { icon: "", pageName: "carl", scriptPath: "carl.py" },
-    ]
+      wrapper.setState({ hideTopBar: false })
 
-    const msg = new ForwardMsg()
-    msg.pagesChanged = new PagesChanged({ appPages })
+      expect(instance.state.hostMenuItems).toStrictEqual([
+        { type: "separator" },
+      ])
+      expect(wrapper.find(MainMenu).prop("hostMenuItems")).toStrictEqual([
+        { type: "separator" },
+      ])
+    })
 
-    const props = getProps()
-    const wrapper = shallow(<App {...props} />)
-    expect(wrapper.find("AppView").prop("appPages")).toEqual([])
+    it("shows hostToolbarItems", () => {
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "SET_TOOLBAR_ITEMS",
+            items: [
+              {
+                key: "favorite",
+                icon: "star.svg",
+              },
+            ],
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
 
-    const instance = wrapper.instance() as App
-    instance.handleMessage(msg)
-    expect(wrapper.find("AppView").prop("appPages")).toEqual(appPages)
+      wrapper.setState({ hideTopBar: false })
+      expect(
+        wrapper.find(ToolbarActions).prop("hostToolbarItems")
+      ).toStrictEqual([
+        {
+          key: "favorite",
+          icon: "star.svg",
+        },
+      ])
+    })
+
+    it("sets hideSidebarNav based on the server config option and host setting", () => {
+      // hideSidebarNav initializes to true.
+      expect(wrapper.find("AppView").prop("hideSidebarNav")).toEqual(true)
 
-    expect(props.hostCommunication.sendMessage).toHaveBeenCalledWith({
-      type: "SET_APP_PAGES",
-      appPages,
+      // Simulate the server ui.hideSidebarNav config option being false.
+      instance.handleNewSession(new NewSession(NEW_SESSION_JSON))
+      expect(wrapper.find("AppView").prop("hideSidebarNav")).toEqual(false)
+
+      // Have the host override the server config option.
+      dispatchEvent(
+        "message",
+        new MessageEvent("message", {
+          data: {
+            stCommVersion: HOST_COMM_VERSION,
+            type: "SET_SIDEBAR_NAV_VISIBILITY",
+            hidden: true,
+          },
+          origin: "https://devel.streamlit.test",
+        })
+      )
+
+      expect(wrapper.find("AppView").prop("hideSidebarNav")).toEqual(true)
     })
-  })
 
-  it("responds to page change requests", () => {
```

Comment:
Is this covered elsewhere?

Reply:
Yup! this test forces update to HostCom HOC state (same thing as receiving `REQUEST_PAGE_CHANGE` message from host) to test `onPageChange` firing. This logic has moved to HostCommMgr -> the parallel test is on line 165 [here](https://github.com/streamlit/streamlit/pull/6746/files#diff-ec570a09c3af41a4893ea3fab14e0a1f2329a33569924b054f7a32916085c322R165)